### PR TITLE
Tweak VV so search results stay after refreshing

### PR DESCRIPTION
**Pull request by quardbreak**
Port of bay port from polaris

### CODE REVIEW

File: code/js/view_variables.js [hunk 1]
```diff
@@ -1,26 +1,52 @@
...
+				var li = children[i];
+				if (li.innerText.toLowerCase().indexOf(filter) == -1) {
+					vars_ol.removeChild(li);
+				}
+			} catch(err) {}
+		}
+	} else {
+		// Remove everything and put back what matches.
+		while (vars_ol.hasChildNodes()) {
+			vars_ol.removeChild(vars_ol.lastChild);
```
Comment (afterthought2) [inline code/js/view_variables.js:24]: Surely there's a better way to do this?

File: code/js/view_variables.js [hunk 2]
```diff
@@ -1,26 +1,52 @@
...
+				var li = complete_list[i];
+				if (!filter || li.innerText.toLowerCase().indexOf(filter) != -1) {
+					vars_ol.appendChild(li);
+				}
+			} catch(err) {}
 		}
 	}
+
+	last_filter = filter;
+	document.cookie="[refid]][cookieoffset]search="+encodeURIComponent(filter);
```
Comment (afterthought2) [inline code/js/view_variables.js:38]: Cookies are questionable in dm, and really unneeded here. If the point is to make the last search persist between window opens, a simple server-side var + Topic call suffices.
```
/holder/var/last_searched = ""

/holder/Topic(href, href_list)
	if(href_list["searched"])
		holder.last_searched = href_list["searched"]
		return

/holder/open_window()
	var/data = "blah[last_searched]blah"
	....
```

File: code/modules/admin/view_variables/view_variables.dm [hunk 3]
```diff
@@ -14,6 +14,8 @@ var/global/list/view_variables_no_assoc = list("verbs", "contents","screen","ima
 	if(!D)
 		return
 
+	var/static/cookieoffset = rand(1, 9999) //to force cookies to reset after the round.
```
Comment (afterthought2) [inline code/modules/admin/view_variables/view_variables.dm:17]: This is pretty terrible; you should be using game id and timing out your cookies.

**Comment by MistakeNot4892** [comment]
Please feel free to reopen this when it has been updated.

**Pull request closed without merging**